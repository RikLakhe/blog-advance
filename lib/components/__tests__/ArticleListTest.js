import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {
    let tree;
    beforeAll(() => {
        const testProps = {
            articles:{
                a:{id:'a'},
                b:{id:'b'}
            },
            articleActions : {
                lookupAuthor : jest.fn(()=>({}))
            }
        };

        tree = renderer.create(<ArticleList {...testProps} />).toJSON();
    });


    it('renders correctly',()=>{
        expect(tree.children.length).toBe(2);

        expect(tree).toMatchSnapshot();
    });
});
