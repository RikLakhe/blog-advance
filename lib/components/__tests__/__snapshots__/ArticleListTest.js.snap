// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "borderBottomColor": "#aaa",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "paddingBottom": 10,
        }
      }
    />
    <div>
      Invalid Date
    </div>
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
  <div>
    <div
      style={
        Object {
          "borderBottomColor": "#aaa",
          "borderBottomStyle": "solid",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "paddingBottom": 10,
        }
      }
    />
    <div>
      Invalid Date
    </div>
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
